packages/coinjoin/src/client/Status.ts

```diff
@@ -110,15 +140,23 @@ export class Status extends EventEmitter {
     private setStatusTimeout() {
         if (!this.enabled) return;
 
-        const nearestDeadline = findNearestDeadline(this.rounds);
-        // TODO: add timeout randomness?
+        const defaultTimeout = STATUS_TIMEOUT[this.mode];
+        const half = defaultTimeout / 2;
+
+        // get deadlines from registered rounds
+        // set minimum deadline to half of defaultTimeout to avoid http request overflow
+        const nearestDeadlines = this.registeredRound
+            .map(r => r.phaseDeadline - Date.now())
+            .map(r => (r > half ? r : half));
```

Comment:
nitpick, slightly easier to read?  
```suggestion
            .map(r => Math.max(r, half));
```

Reply:
sure, fixed rebased and forcepushed